### PR TITLE
Add a walkthrough experience for the VSCode extension

### LINKED ISSUE #1404: Add a Getting Started page to the Azure Developer CLI extension in VS Code
When we detect an `azure.yaml` is present in the root of the current workspace, we should show users who have the azd extension installed a Getting Started walkthrough. This walkthrough should have a big shiny button that tells them that they can provision and deploy in a single step. On clicking said button, we should open the command palette or the `azure.yaml` context menu and show `azd up`.

Pieces of information to figure out:
1. Title and description
2. Steps
	1. Title, description. Description can have up to a single button to run commands, or several that will render as hyperlinks.
	2. Optional media (Markdown, GIF, jpg)
	3. Optional completion criteria (what makes the box checked)
	4. Optional filters for whether or not it should show the step

### PR DESCRIPTION
**Pull request by bwateratmsft**
Closes #1404

image
Install step. This step will be hidden ~1 second after extension activation if the CLI is already installed. Step is marked completed when they install, or automatically if it was already installed before opening.

image
Sign in step. Similarly, will be hidden ~1 second after activation if already logged in. Step is marked completed when they sign in, or automatically if they were already signed in when they open it.

image
Scaffold step. Similarly, will be hidden ~1 second after activation if an azure.yaml (or azure.yml) file exists _at the root of the workspace_--scoped as such for performance. Step is marked completed when they init, or if project was scaffolded before opening.

image
Up step. Step is marked completed when the user does up, provision, deploy, or infra-create; though the button specifically does up. The "Command Palette" link will pre-type "> Azure Developer: up" to immediately show the relevant command.

image
Learn More step. Step is marked completed when it is clicked on to. The "Command Palette" link will pre-type "> Azure Developer: " to immediately show the relevant commands.

image
Light theme of Learn More step (scaffold and up steps work the same way)

### CODE REVIEW

**Review by karolz-ms — APPROVED** [review]

File: ext/vscode/package.json [hunk 1]
```diff
@@ -364,7 +370,81 @@
                     }
                 ]
             }
-        }
+        },
+        "walkthroughs": [
+            {
+                "id": "azd.start",
```
Comment (savannahostrowski) [inline ext/vscode/package.json:376]: Is this id what will come through in telemetry? Or unrelated?

Does this have telemetry attached to it?

File: ext/vscode/package.nls.json [hunk 2]
```diff
@@ -90,5 +92,23 @@
     "azure-dev.surveys.surveyOne.prompt": "Can you please take 2 minutes to tell us how the Azure Developer CLI is working for you?",
     "azure-dev.surveys.surveyOne.button.take": "Take survey",
     "azure-dev.surveys.surveyOne.button.never": "Don't ask again",
-    "azure-dev.surveys.surveyOne.button.later": "Remind me later"
+    "azure-dev.surveys.surveyOne.button.later": "Remind me later",
+
+    "azure-dev.walkthroughs.start.title": "Deploy your App with the Azure Developer CLI",
+    "azure-dev.walkthroughs.start.description": "Your quick guide to set up, provision, and deploy a new app with the AZD CLI.",
```
Comment (savannahostrowski) [inline ext/vscode/package.nls.json:98]: Can we do lowercase `azd`?

File: ext/vscode/package.json (hunk 1 above)
Comment (bwateratmsft) [reply ext/vscode/package.json:376]: I believe there is core VSCode telemetry for walkthroughs. The short ID is to get around length limits on treatment variables in EXP.

File: ext/vscode/package.nls.json [hunk 3]
```diff
@@ -90,5 +92,23 @@
...
+    "azure-dev.walkthroughs.start.steps.install.description": "Install the Azure Developer CLI (**azd**) to get started with Azure development.\n[Install](command:azure-dev.commands.cli.install)",
+
+    "azure-dev.walkthroughs.start.steps.login.title": "Sign In to your Azure Account",
+    "azure-dev.walkthroughs.start.steps.login.description": "Sign in or create an Azure account to successfully run **azd** commands.\n[Sign In](command:azure-dev.commands.cli.login)\nTip: Clicking the **Sign In** button may open a new browser window with a prompt to input your Azure credentials.",
+
+    "azure-dev.walkthroughs.start.steps.scaffold.title": "Scaffold the Code",
+    "azure-dev.walkthroughs.start.steps.scaffold.description": "View the [Awesome azd template gallery](https://azure.github.io/awesome-azd/) to find a project with the infrastructure that's right for you, or select a template directly from the Command Palette (``azd init``).\n[Select a template and initialize your project](command:azure-dev.commands.cli.init)\nTip: To use a template from the Awesome azd site, copy its GitHub repo URL and enter it when prompted during ``azd init``.",
```
Comment (savannahostrowski) [inline ext/vscode/package.nls.json:107]: Can we reword "View the [Awesome azd template gallery](https://azure.github.io/awesome-azd/) to find a project with the infrastructure that's right for you"

To something like:

"View the [Awesome azd template gallery](https://azure.github.io/awesome-azd/) to find a template with the infrastructure that's right for your app" 

I want to make sure that the value prop of the the templates (IaC + editor configs etc) isn't lost here. Trying to think about how to frame this in the most compelling way.

@gkulin thoughts?

**Review by savannahostrowski — COMMENTED** [review]
A couple of copy nits but other than that, LGTM.

File: ext/vscode/package.json (hunk 1 above)
Comment (bwateratmsft) [reply ext/vscode/package.json:376]: Yep, I confirmed with Harald, VSCode will capture it for us.

Comment (savannahostrowski) [reply ext/vscode/package.json:376]: Great!

**Review by ellismg — APPROVED** [review]

File: ext/vscode/package.json [hunk 4]
```diff
@@ -364,7 +370,81 @@
...
+                            "onCommand:azure-dev.commands.cli.login"
+                        ],
+                        "when": "hideAzdLoginStep != true"
+                    },
+                    {
+                        "id": "azd.scaffold",
+                        "title": "%azure-dev.walkthroughs.start.steps.scaffold.title%",
+                        "description": "%azure-dev.walkthroughs.start.steps.scaffold.description%",
+                        "media": {
+                            "svg": "resources/walkthrough/azdScaffold.svg",
```
Comment (bwateratmsft) [inline ext/vscode/package.json:412]: Awaiting a revised SVG

**Review by gkulin — APPROVED** [review]

**Pull request merged**